ledokku-bootstrap.sh

```diff
@@ -1,52 +1,174 @@
 #!/bin/bash
-# This script will install ledokku on your server
-# You need to already have dokku installed in order to be able to run it
+
+# This script is used to configure the essentials of t2d.
+## 1 => Check whether the program/application, "Whiptails" exists or not.
+## 2 => Making sure that the script is runing with root permissions.
+## 3 => Update and Upgrade the VPS.
+## 4 => Check whether the program/application, "Dokku" was Installed or not.
+## 5 => Upgrading dokku to the latest version.
+## 6 => Check Whether Plugins are installed or not (Redis, Postgres)
+## 7 => Install Redis and Postgress, if they were not installed
+## 7 => Install Ledokku
+## ? => Update Script (Will be added later)
+
 set -e
 
-# Check that dokku is installed on the server
-ensure-dokku() {
-  if ! command -v dokku &> /dev/null
-  then
-      echo "dokku is not installed, please follow our getting started guide first"
-      echo "https://www.ledokku.com/docs/getting-started"
+function define-colors {
+
+  # Using tput will eliminate the usage of "-e" in echo, and can be used anywhere
+  ## Color Palet
+  ## Should exist in every script
+  RED="$(tput setaf 1)" # ${RED}
+  GREEN="$(tput setaf 2)" # ${GREEN}
+  YELLOW="$(tput setaf 3)" # ${YELLOW}
+  BLUE="$(tput setaf 123)" # ${BLUE}
+  END="$(tput setaf 7)" # ${END
+}
+
+function system-info() {
+  
+  # Finding Information about your device
+  ## Basic VPS info
+  ## Should exist in every script
+  # IP="$(ifconfig | grep broadcast | awk '{print $2}')"
+  DOKKU_SSH_HOST=$(curl -4 ifconfig.co)
+  OS=$( $(compgen -G "/etc/*release" > /dev/null) && cat /etc/*release | grep ^NAME | tr -d 'NAME="' || echo "${OSTYPE//[0-9.]/}")
+}
+
+function check-whiptail() {
+  
+  # Checking if whiptail is available or not
+  if which whiptail >/dev/null; then
+      echo "${GREEN}whiptail exists${END}"
+      # Continue the script
+  else
+      echo "${RED}whiptail does not exist${END}"
+      echo "Install whiptail and re-run the script, your OS is ${OS}"
       exit
+      # As I already know the OS, I can also automate this process, but will save it for later.
+      # This is just a matter of finiding all the possible OS people might use and writing a if statement.
```

Comment:
Typo `finiding `

Reply:
let's figure out all the typos and edit at the end rather than making simultaneous commits for individual typos.

Reply:
#### Typo, Edit 01
finiding => finding
a if   => an if